File: backend/app/schemas/video.py
from pydantic import BaseModel, ConfigDict
from typing import Optional, List
from datetime import datetime
from uuid import UUID
from app.models.video import VideoStatus

class VideoBase(BaseModel):
    title: str
    description: Optional[str] = None

class VideoResponse(VideoBase):
    id: UUID
    user_id: UUID
    file_path: str
    status: VideoStatus
    transcription: Optional[str] = None
    ai_tags: Optional[List[str]] = None
    views_count: int
    likes_count: int
    created_at: datetime
    
    model_config = ConfigDict(from_attributes=True)

class CreatorInfo(BaseModel):
    id: UUID
    first_name: Optional[str] = None
    last_name: Optional[str] = None
    trust_level: int
    
class VideoFeedResponse(VideoResponse):
    creator: CreatorInfo
    
    model_config = ConfigDict(from_attributes=True)

class VideoViewCreate(BaseModel):
    session_id: str
    watched_seconds: int = 0
    completed: bool = False

class VideoReportCreate(BaseModel):
    reason: str

class VideoStatsResponse(BaseModel):
    views: int
    likes: int
    liked: bool

File: frontend/src/types/video.ts
export type VideoStatus = 'UPLOADING' | 'PROCESSING' | 'PENDING_CONSENT' | 'PUBLISHED' | 'REJECTED';

export interface Video {
  id: string;
  user_id: string;
  title: string;
  description?: string;
  file_path: string;
  status: VideoStatus;
  transcription?: string;
  ai_tags?: string[];
  views_count: number;
  likes_count: number;
  created_at: string;
}

export interface CreatorInfo {
  id: string;
  first_name?: string;
  last_name?: string;
  trust_level: number;
}

export interface VideoFeedResponse extends Video {
  creator: CreatorInfo;
}

// ── Dashboard Analytics ────────────────────────────────────────────────────

export interface VideoAnalytics {
  id: string;
  title: string;
  description?: string;
  status: VideoStatus;
  views_count: number;
  likes_count: number;
  avg_watch_seconds: number;
  completion_rate: number;  // % des viewers qui regardent jusqu'au bout
  ai_tags: string[];
  created_at: string;
}

export interface TalentAnalytics {
  // KPIs globaux
  total_views: number;
  total_likes: number;
  engagement_rate: number;  // likes / views * 100
  total_videos: number;
  published_videos: number;
  pending_videos: number;
  rejected_videos: number;
  times_saved_as_favorite: number;
  recruiter_contacts: number;
  profile_views: number;   // préparé pour le futur
  // Détail par vidéo
  videos: VideoAnalytics[];
}
